virtual_rainforest/models/abiotic/abiotic_model.py

```diff
@@ -238,5 +249,38 @@ def update(self, time_index: int, **kwargs: Any) -> None:
 
         self.data.add_from_dict(output_dict=wind_output)
 
+        # Soil energy balance
+        topsoil_layer_index = next(
+            i
+            for i, v in enumerate(self.data["soil_temperature"].layer_roles)
+            if v == "soil"
+        )
```

Comment:
I think this is more efficient as `layer_roles.index("soil")` - since we're explicitly after the first soil layer?

I'm beginning to think that this index should be part of the new LayerStructure - you and @jacobcook1995  keep calculating it! So then you could just use `self.layer_structure.topsoil_layer_index`

Reply:
I agree, this might be a good idea. For now, I make the change you suggested and will update when we roll out the new LayerStructure